contrib/hyperscan-cmake/CMakeLists.txt

```diff
@@ -0,0 +1,252 @@
+            ${LIBRARY_DIR}/src/rose/program_runtime.c
+            ${LIBRARY_DIR}/src/rose/rose_build_add_mask.cpp
+            ${LIBRARY_DIR}/src/rose/rose_build_add.cpp
+            ${LIBRARY_DIR}/src/rose/rose_build_anchored.cpp
+            ${LIBRARY_DIR}/src/rose/rose_build_bytecode.cpp
+            ${LIBRARY_DIR}/src/rose/rose_build_castle.cpp
+            ${LIBRARY_DIR}/src/rose/rose_build_compile.cpp
+            ${LIBRARY_DIR}/src/rose/rose_build_convert.cpp
+            ${LIBRARY_DIR}/src/rose/rose_build_dedupe.cpp
+            ${LIBRARY_DIR}/src/rose/rose_build_engine_blob.cpp
+            ${LIBRARY_DIR}/src/rose/rose_build_exclusive.cpp
+            ${LIBRARY_DIR}/src/rose/rose_build_groups.cpp
+            ${LIBRARY_DIR}/src/rose/rose_build_infix.cpp
+            ${LIBRARY_DIR}/src/rose/rose_build_instructions.cpp
+            ${LIBRARY_DIR}/src/rose/rose_build_lit_accel.cpp
+            ${LIBRARY_DIR}/src/rose/rose_build_long_lit.cpp
+            ${LIBRARY_DIR}/src/rose/rose_build_lookaround.cpp
+            ${LIBRARY_DIR}/src/rose/rose_build_matchers.cpp
+            ${LIBRARY_DIR}/src/rose/rose_build_merge.cpp
+            ${LIBRARY_DIR}/src/rose/rose_build_misc.cpp
+            ${LIBRARY_DIR}/src/rose/rose_build_program.cpp
+            ${LIBRARY_DIR}/src/rose/rose_build_role_aliasing.cpp
+            ${LIBRARY_DIR}/src/rose/rose_build_scatter.cpp
+            ${LIBRARY_DIR}/src/rose/rose_build_width.cpp
+            ${LIBRARY_DIR}/src/rose/rose_in_util.cpp
+            ${LIBRARY_DIR}/src/rose/stream.c
+            ${LIBRARY_DIR}/src/runtime.c
+            ${LIBRARY_DIR}/src/scratch.c
+            ${LIBRARY_DIR}/src/smallwrite/smallwrite_build.cpp
+            ${LIBRARY_DIR}/src/som/slot_manager.cpp
+            ${LIBRARY_DIR}/src/som/som_runtime.c
+            ${LIBRARY_DIR}/src/som/som_stream.c
+            ${LIBRARY_DIR}/src/stream_compress.c
+            ${LIBRARY_DIR}/src/util/alloc.cpp
+            ${LIBRARY_DIR}/src/util/charreach.cpp
+            ${LIBRARY_DIR}/src/util/clique.cpp
+            ${LIBRARY_DIR}/src/util/compile_context.cpp
+            ${LIBRARY_DIR}/src/util/compile_error.cpp
+            ${LIBRARY_DIR}/src/util/cpuid_flags.c
+            ${LIBRARY_DIR}/src/util/depth.cpp
+            ${LIBRARY_DIR}/src/util/fatbit_build.cpp
+            ${LIBRARY_DIR}/src/util/multibit_build.cpp
+            ${LIBRARY_DIR}/src/util/multibit.c
+            ${LIBRARY_DIR}/src/util/report_manager.cpp
+            ${LIBRARY_DIR}/src/util/simd_utils.c
+            ${LIBRARY_DIR}/src/util/state_compress.c
+            ${LIBRARY_DIR}/src/util/target_info.cpp
+            ${LIBRARY_DIR}/src/util/ue2string.cpp
+        )
+
+        add_library (hyperscan ${SRCS})
+
+        target_compile_definitions (hyperscan PUBLIC USE_HYPERSCAN=1)
+        target_compile_options (hyperscan
+            PRIVATE -g0 -march=corei7 # library has too much debug information
+            PUBLIC -Wno-documentation
```

Comment:
This is wrong. Warning suppression can never be public.

Reply:
There is no other way, then fixing it in the code - it's a problem in public headers. Fortunately, it's in our ClickHouse-Extras. Will fix it there.

Reply:
Actually, there are some other potential ways - let's discuss in TG.